edn/src/namespaced_name.rs

```diff
@@ -0,0 +1,220 @@
+// Copyright 2016 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use std::cmp::{Ord, PartialOrd, Ordering};
+use std::fmt;
+
+#[cfg(feature = "serde_support")]
+use serde::{
+    de::{self, Deserialize, Deserializer},
+    ser::{Serialize, Serializer}
+};
+
+// Data storage for both NamespacedKeyword and NamespacedSymbol. This gets it's own module because
+// modifying it's fields is `unsafe`, and so we'd like to minimize the amount of code that can
+// touch it.
+#[derive(Clone, Eq, Hash, PartialEq)]
+pub struct NamespacedName {
+    // The bytes that make up the namespace followed directly by those
```

Comment:
I would really like this to include any slash, so that we internally store `:foo/bar` or `foo/bar` or whatever.  That gives us a hope of implementing `AsRef<str>`.  If we don't store the leading colon and/or internal slash, that's out.

Reply:
Filed #692.